core/src/main/java/io/questdb/griffin/engine/join/AsOfJoinNoKeyFastRecordCursorFactory.java

```diff
@@ -0,0 +1,369 @@
+            }
+            return slaveCursor.newSymbolTable(columnIndex - columnSplit);
+        }
+
+        @Override
+        public long size() {
+            return masterCursor.size();
+        }
+
+        @Override
+        public void toTop() {
+            slaveTimestamp = Long.MIN_VALUE;
+            slaveFrameIndex = -1;
+            slaveFrameRow = Long.MIN_VALUE;
+            record.hasSlave(false);
+            masterCursor.toTop();
+            slaveCursor.toTop();
+            isMasterHasNextPending = true;
+            isSlaveOpenPending = false;
+            isSlaveForwardScan = true;
+        }
+
+        private boolean linearScan(TimeFrame frame, long masterTimestamp) {
+            final long scanHi = Math.min(slaveFrameRow + lookahead, frame.getRowHi());
+            while (slaveFrameRow < scanHi) {
+                slaveCursor.recordAt(slaveRecA, Rows.toRowID(slaveFrameIndex, slaveFrameRow));
+                slaveTimestamp = slaveRecA.getTimestamp(slaveTimestampIndex);
+                if (slaveTimestamp > masterTimestamp) {
+                    return true;
+                }
+                record.hasSlave(true);
+                slaveCursor.recordAt(slaveRecB, Rows.toRowID(slaveFrameIndex, slaveFrameRow));
+                slaveFrameRow++;
+            }
+            return false;
+        }
+
+        private void nextSlave(long masterTimestamp) {
+            final TimeFrame frame = slaveCursor.getTimeFrame();
+            while (true) {
+                if (frame.isOpen() && frame.getIndex() == slaveFrameIndex) {
+                    // Scan a few rows to speed up self-join/identical tables cases.
+                    if (linearScan(frame, masterTimestamp)) {
+                        return;
+                    }
+                    if (slaveFrameRow < frame.getRowHi()) {
+                        // Fallback to binary search.
+                        // Find the last value less or equal to the master timestamp.
+                        long foundRow = binarySearch(
+                                masterTimestamp,
+                                slaveFrameRow,
+                                frame.getRowHi() - 1,
+                                slaveCursor,
+                                slaveFrameIndex,
+                                slaveRecA,
+                                slaveTimestampIndex
+                        );
+                        if (foundRow < slaveFrameRow) {
+                            // All searched timestamps are greater than the master timestamp.
+                            // Linear scan must have found the row.
+                            return;
+                        }
+                        slaveFrameRow = foundRow;
+                        record.hasSlave(true);
+                        slaveCursor.recordAt(slaveRecB, Rows.toRowID(slaveFrameIndex, slaveFrameRow));
+                        slaveTimestamp = slaveRecB.getTimestamp(slaveTimestampIndex);
```

Comment:
as per the findings slaveTimestamp is next after the matching slaveTimestamp value, this should be modified to return be +1 to the value on the method return

Reply:
This is not necessary for correctness, but for code consistency. Changed in e914964f0277695e83bd3ea8800b0d4b53f18361